### PR TITLE
Feature/bb 3230 docdata latest payment

### PR DESCRIPTION
**Pull request by eodolphi**
To check the status of a payments, we should look at the totals, instead of looking at the status of the first payment.

This branch fixes that, reenables the storing of transactions, and adds a bunch of tests for payments status response edge cases.

### CODE REVIEW

**Comment by aethemba** [comment]
This will pass when you merge the fixed PR of @gannetson  that includes a simple text change.

File: bluebottle/payments_docdata/adapters.py [hunk 1]
```diff
@@ -285,13 +266,40 @@ def check_payment_status(self):
...
+                # Everything was captured
+                status = StatusDefinition.SETTLED
+            elif int(totals.totalAcquirerApproved) + int(totals.totalAcquirerPending) + int(totals.totalShopperPending) == int(totals.totalRegistered):
+                # Everything was authorized
+                status = StatusDefinition.AUTHORIZED
+            else:
+                # Anything else (Partly captured, partly charged back, etc)
+                status = StatusDefinition.UNKNOWN
+
+        if self.payment.status != status:
```
Comment (gannetson) [inline bluebottle/payments_docdata/adapters.py]: Maybe we should always update this, even if the status doesn't change?
Are you sure this doesn't change while status doesn't?

Comment (eodolphi) [inline bluebottle/payments_docdata/adapters.py]: I was wondering about that to. Kept it the same for now. If this is not a performance problem, then yes let's change this just to be sure the data is correct.

**Comment by gannetson** [comment]
@eodolphi could you merge saas branch so this can be pulled cleanly?

**Comment by eodolphi** [comment]
@gannetson Done!
